### PR TITLE
Aircraft queue

### PR DESCRIPTION
**Pull request by macgyver**
todo: 
- [ ] fixup tests
- [ ] deploy latest to gh pages

### CODE REVIEW

File: app/components/aircraft-queue.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          dequeueCandidates = largeAircraft;
+        }
+
+        // finally pick the earliest remaining object
+        dequeuedAircraft = dequeueCandidates.get('firstObject');
+      }
+
+      this.get('aircrafts').removeObject(dequeuedAircraft);
+      this.set('dequeuedAircraft', dequeuedAircraft);
+    }
```
Comment (macgyver) [inline app/components/aircraft-queue.js]: Let's refactor this to use 4 queues.

File: app/components/aircraft-queue.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import Ember from 'ember';
+const {computed} = Ember;
+import {TYPES as AIRCRAFT_TYPES, SIZES as AIRCRAFT_SIZES} from '../models/aircraft';
+
+// poor man's method of generating unique numbers,
+// purely to assist human readers in identifying airplanes
+const numbers = [];
+function makeNumber() {
```
Comment (nastycoder) [inline app/components/aircraft-queue.js]: I found [this](https://developer.mozilla.org/en-US/docs/Web/API/Window/crypto) the other day and figured I'd share.

So your ```makeNumber``` function would be simplified to:

```
function makeNumber() {
  return window.crypto.getRandomValues(new Uint32Array(1))[0];
}
```

Not a game changer, just a neat trick I learned a few days ago.

Comment (macgyver) [reply app/components/aircraft-queue.js]: Haha, fun - thanks for the tip. Just curious, how did you stumble on this repo and decide to comment?

Comment (nastycoder) [reply app/components/aircraft-queue.js]: I'm a senior engineer at Bublup. Alberto asked me to take a look.

Comment (macgyver) [reply app/components/aircraft-queue.js]: Aaaaah that makes more sense than "random open source do-gooder/ web crypto enthusiast"

**Pull request closed without merging**